pandas/core/indexes/datetimes.py

```diff
@@ -1089,6 +1080,11 @@ def slice_indexer(self, start=None, end=None, step=None, kind=None):
     # --------------------------------------------------------------------
     # Wrapping DatetimeArray
 
+    @property
```

Comment:
shouldn't this always be cached? maybe call this _impl instead

Reply:
Yes, but for the proof of concept I didn't want to futz with it.  If this is a direction we want to move forward with I'll do this.